graphs/graph_list.py

```diff
@@ -1,44 +1,154 @@
+            elif source_vertex in self.adj_list:
+                self.adj_list[source_vertex].append(destination_vertex)
+                self.adj_list[destination_vertex] = [source_vertex]
+            # if only destination vertex is present in adjacency list, add source vertex
+            # to destination vertex list of adjacent vertices, then create a new vertex
+            # with source vertex as key and assign a list containing the source vertex
+            # as it's first adjacent vertex.
+            elif destination_vertex in self.adj_list:
+                self.adj_list[destination_vertex].append(source_vertex)
+                self.adj_list[source_vertex] = [destination_vertex]
+            # if both source vertex and destination vertex are not present in adjacency
+            # list, create a new vertex with source vertex as key and assign a list
+            # containing the destination vertex as it's first adjacent vertex also
+            # create a new vertex with destination vertex as key and assign a list
+            # containing the source vertex as it's first adjacent vertex.
+            else:
+                self.adj_list[source_vertex] = [destination_vertex]
+                self.adj_list[destination_vertex] = [source_vertex]
+
+        # For directed Graphs
         else:
-            self.adj_list[from_vertex] = [to_vertex]
+            # if both source vertex and destination vertex are present in adjacency
+            # list, add destination vertex to source vertex list of adjacent vertices.
+            if source_vertex in self.adj_list and destination_vertex in self.adj_list:
+                self.adj_list[source_vertex].append(destination_vertex)
+            # if only source vertex is present in adjacency list, add destination
+            # vertex to source vertex list of adjacent vertices and create a new vertex
+            # with destination vertex as key, which has no adjacent vertex
+            elif source_vertex in self.adj_list:
+                self.adj_list[source_vertex].append(destination_vertex)
+                self.adj_list[destination_vertex] = []
+            # if only destination vertex is present in adjacency list, create a new
+            # vertex with source vertex as key and assign a list containing destination
+            # vertex as first adjacent vertex
+            elif destination_vertex in self.adj_list:
+                self.adj_list[source_vertex] = [destination_vertex]
+            # if both source vertex and destination vertex are not present in adjacency
+            # list, create a new vertex with source vertex as key and a list containing
+            # destination vertex as it's first adjacent vertex. Then create a new vertex
+            # with destination vertex as key, which has no adjacent vertex
+            else:
+                self.adj_list[source_vertex] = [destination_vertex]
+                self.adj_list[destination_vertex] = []
```

Comment:
It would be a powerful addition if this function returned the new graph to enable the fluent syntax:
`graph_adjacency_list = GraphAdjacencyList().add_edge(0, 1).add_edge(1, 3)`